pkg/athena/datasource.go

```diff
@@ -51,6 +57,18 @@ func (s *AthenaDatasource) Connect(config backend.DataSourceInstanceSettings, qu
 		return nil, err
 	}
 
+	// athena datasources require a region to establish a connection, we use a default region if none was provided
+	if args["region"] == "" {
+		if config.JSONData != nil && len(config.JSONData) > 1 {
+			settings := JsonSettings{}
+			err := json.Unmarshal(config.JSONData, &settings)
+			if err != nil {
+				return nil, err
+			}
+			args["region"] = settings.DefaultRegion
+		}
+	}
+
```

Comment:
in line 81 (getAPI), you also need modify `options` (which is the same than `args` here), and forge an object that only uses the region, and avoid the rest of options (which are not used to obtain an API instance).